src/eddymotion/estimator.py

```diff
@@ -105,45 +104,32 @@ def fit(
                         # generate a synthetic dw volume for the test gradient
                         predicted = dwmodel.predict(data_test[1])
 
-                        tmpdir = Path(tmpdir)
-                        moving = tmpdir / "moving.nii.gz"
-                        fixed = tmpdir / "fixed.nii.gz"
-                        _to_nifti(data_test[0], dwdata.affine, moving)
-                        _to_nifti(
-                            predicted,
-                            dwdata.affine,
-                            fixed,
-                            clip=reg_target_type == "dwi",
-                        )
+                        moving_nii = nb.Nifti1Image(data_test[0], dwdata.affine)
+                        moving_ants = ants.from_nibabel(moving_nii)
+
+                        fixed_nii = nb.Nifti1Image(predicted, dwdata.affine)
+                        fixed_ants = ants.from_nibabel(fixed_nii)
+
+                        registration_kwargs = dict(fixed=fixed_ants, moving=moving_ants)
 
-                        registration = Registration(
-                            terminal_output="file",
-                            from_file=pkg_fn(
-                                "eddymotion",
-                                f"config/dwi-to-{reg_target_type}_level{i_iter}.json",
```

Comment:
One question to mull over is how do we pass this config to `ants.registration`? Right now, it's no longer being passed, and it uses the default settings of that function instead.